File: app/src/main/java/com/mohamedtaha/imagine/ui/activities/YoutubeActivity.java
package com.mohamedtaha.imagine.ui.activities;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import com.mohamedtaha.imagine.Config;
import com.google.android.youtube.player.YouTubeBaseActivity;
import com.google.android.youtube.player.YouTubeInitializationResult;
import com.google.android.youtube.player.YouTubePlayer;
import com.google.android.youtube.player.YouTubePlayerView;
import com.mohamedtaha.imagine.R;

public class YoutubeActivity extends YouTubeBaseActivity implements YouTubePlayer.OnInitializedListener {
    private static final int RECOVERY_REQUEST = 1;
    private static final String LINK_YOUTUBE = "SMNHZR1u6KQ";
    private YouTubePlayerView youTubeView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_youtube);
        youTubeView = findViewById(R.id.Youtube_View);
        youTubeView.initialize(Config.YOUTUBE_API_KEY, this);
    }

    @Override
    public void onInitializationSuccess(YouTubePlayer.Provider provider, YouTubePlayer youTubePlayer, boolean b) {
        if (!b) {
            youTubePlayer.cueVideo(LINK_YOUTUBE);
            Log.d("TAG", "Play");
        }
    }

    @Override
    public void onInitializationFailure(YouTubePlayer.Provider provider, YouTubeInitializationResult youTubeInitializationResult) {
        if (youTubeInitializationResult.isUserRecoverableError()) {
            youTubeInitializationResult.getErrorDialog(this, RECOVERY_REQUEST).show();
        } else {
            String error = String.format(getString(R.string.player_error), youTubeInitializationResult.toString());
            Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == RECOVERY_REQUEST) {
            // Retry initialization if user performed a recovery action
            getYoutubePlayerPRovider().initialize(Config.YOUTUBE_API_KEY, this);
        }
        super.onActivityResult(requestCode, resultCode, data);
    }

    protected YouTubePlayer.Provider getYoutubePlayerPRovider() {
        return youTubeView;
    }
}

File: app/src/main/java/com/mohamedtaha/imagine/ui/youtube/YoutubeActivity.kt
package com.mohamedtaha.imagine.ui.youtube

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.asLiveData
import com.google.android.youtube.player.YouTubeInitializationResult
import com.google.android.youtube.player.YouTubePlayer
import com.google.android.youtube.player.YouTubePlayerSupportFragment
import com.mohamedtaha.imagine.BuildConfig.YOUTUBE_API_KEY
import com.mohamedtaha.imagine.R
import com.mohamedtaha.imagine.databinding.ActivityYoutubeBinding
import com.mohamedtaha.imagine.datastore.DataStoreViewModel
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class YoutubeActivity : AppCompatActivity(), YouTubePlayer.OnInitializedListener {
    private lateinit var binding: ActivityYoutubeBinding
    private val datStoreViewModel: DataStoreViewModel by viewModels()

    private var channelLink = ""
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
         binding = ActivityYoutubeBinding.inflate(layoutInflater)
        setContentView(binding.root)

//        val fragment: YouTubePlayerSupportFragment =
//            (YouTubePlayerSupportFragment) supportFragmentManager!!.findFragmentById (R.id.youtubeFragment)
//        fragment.initialize(Config.YOUTUBE_API_KEY, this)
        //   binding.youtubeFragment.initialize(Config.YOUTUBE_API_KEY, this)
        datStoreViewModel.youTubeChannel.asLiveData().observe(this@YoutubeActivity) {
            channelLink = it
        }

        val youTubePlayerFragment = supportFragmentManager
            .findFragmentById(R.id.youtubeFragment) as YouTubePlayerSupportFragment? ?: return
        youTubePlayerFragment.initialize(YOUTUBE_API_KEY, this)
    }

    override fun onInitializationSuccess(
        provider: YouTubePlayer.Provider,
        youTubePlayer: YouTubePlayer,
        b: Boolean
    ) {
        if (!b) {
            youTubePlayer.setPlayerStyle(YouTubePlayer.PlayerStyle.MINIMAL)
            youTubePlayer.cueVideo(channelLink)
            Log.d("Play", "Play")
        }
    }

    override fun onInitializationFailure(
        provider: YouTubePlayer.Provider,
        youTubeInitializationResult: YouTubeInitializationResult
    ) {
        if (youTubeInitializationResult.isUserRecoverableError) {
            youTubeInitializationResult.getErrorDialog(this, RECOVERY_REQUEST).show()
        } else {
            val error = String.format(getString(R.string.player_error), youTubeInitializationResult)
            Toast.makeText(this, error, Toast.LENGTH_SHORT).show()
        }
    }

//    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
//        if (requestCode == RECOVERY_REQUEST) {
//            // Retry initialization if user performed a recovery action
//            youtubePlayerProvider.initialize(Config.YOUTUBE_API_KEY, this)
//        }
//        super.onActivityResult(requestCode, resultCode, data)
//    }
//
//    private val youtubePlayerProvider: YouTubePlayer.Provider
//        get() = binding.YoutubeView

    companion object {
        private const val RECOVERY_REQUEST = 1
    }
}